src/main/java/net/hypixel/nerdbot/curator/ForumChannelCurator.java

```diff
@@ -0,0 +1,92 @@
+                    List<Message> allMessages = thread.getIterableHistory().complete(true);
+                    Message firstPost = allMessages.get(allMessages.size() - 1);
+                    Emoji agreeEmoji = getBot().getJDA().getEmojiById(getBot().getConfig().getEmojis().getAgree());
+                    Emoji disagreeEmoji = getBot().getJDA().getEmojiById(getBot().getConfig().getEmojis().getDisagree());
+
+                    if (agreeEmoji == null || disagreeEmoji == null) {
+                        log("Couldn't find the agree or disagree emoji, time to yell!");
+                        ChannelManager.getLogChannel().sendMessage(Users.getUser(Users.AERH).getAsMention() + " I couldn't find the agree or disagree emoji for some reason, check logs!").queue();
+                        break;
+                    }
+
+                    // Add the reactions if they're missing
+                    // Theoretically the agree reaction should be there by default but lets check anyway just in case
+                    MessageReaction agreeReaction = firstPost.getReaction(agreeEmoji);
+                    MessageReaction disagreeReaction = firstPost.getReaction(disagreeEmoji);
+
+                    if (agreeReaction == null) {
+                        log("Adding a missing agree reaction to post " + firstPost.getId());
+                        firstPost.addReaction(agreeEmoji).complete();
+                    }
+
+                    if (disagreeReaction == null) {
+                        log("Adding a missing disagree reaction to post " + firstPost.getId());
+                        firstPost.addReaction(disagreeEmoji).complete();
+                    }
+
+                    if (agreeReaction.getCount() < getBot().getConfig().getMinimumThreshold()) {
+                        log("Post " + firstPost.getId() + " doesn't have enough agree reactions, skipping...");
+                        continue;
+                    }
+
+                    // Get the ratio of reactions and greenlight it if it's over the threshold
+                    double ratio = getRatio(agreeReaction.getCount(), disagreeReaction.getCount());
+                    if (ratio >= getBot().getConfig().getPercentage()) {
+                        GreenlitMessage greenlitMessage = GreenlitMessage.builder()
+                                .agrees(agreeReaction.getCount())
+                                .disagrees(disagreeReaction.getCount())
+                                .messageId(firstPost.getId())
+                                .build();
+
+                        log("Greenlighting thread '" + thread.getName() + "' (Thread ID: " + thread.getId() + ") with a ratio of " + ratio + "%");
+                        thread.getManager().setAppliedTags(new ForumTagImpl(getBot().getConfig().getTags().getGreenlit())).complete();
+                        output.add(greenlitMessage);
+                    }
+                } catch (RateLimitedException exception) {
```

Comment:
double checking, but if this exception is caught, when do we curate this message again?

Reply:
Next time the curator is running, so ideally every interval specified in the config or if it's manually started with the command